### PR TITLE
Closes #401 No longer implementing our own `rlang::is_named()` and not exporting it

### LINKED ISSUE #401: Re-export `rlang::is_named()` ?
We re implementing our own version of `is_named()` that masks `rlang::is_named()` when both packages are loaded. it seems to me that it's doing the same thing. Perhaps we should just reexport the rlang function instead.

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/dev/CONTRIBUTING.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #401 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 ## Breaking Changes
 
-* `renv` and related files have been removed (#360)
+* `renv` and related files have been removed. (#360)
+
+* No longer exporting `is_named()` function. (#401)
```
Comment (bundfussr) [inline NEWS.md:11]: I think we should follow our deprecation process.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: NEWS.md (hunk 1 above)
Comment (ddsjoberg) [reply NEWS.md:11]: Thinking a bit more about the logistics here. I don't think there is a way to import the function from rlang and deprecate its exported equivalent at the same time.

I see two options:
1. Leave it as is as a breaking change, and remove the function from export. I can't imagine users relying on an exported version of `is_named()` when there is an rlang equivalent.
2. We can deprecate our implementation of `is_named()`, but we cannot import `is_named()`. We would need to change is call to the function adding the rlang namespace.

Comment (bms63) [reply NEWS.md:11]: I say we go for option 1 and just make a note of it in the news.

I did a quick search of all pharmaverse repos in github and don't see this function being used except in `admiral` everyone else uses `rlang::is_named()` https://github.com/search?q=org%3Apharmaverse+is_named&type=code

Comment (ddsjoberg) [reply NEWS.md:11]: ok @bms63 sounds good

**Review by bundfussr — COMMENTED** [review]
In line 965 of `assertions.R`
```
any_unnamed <- length(arg) > 0L && !rlang::is_named(arg)
```
the `rlang::` should be removed.

**Comment by ddsjoberg** [comment]
> In line 965 of `assertions.R`
> 
> ```
> any_unnamed <- length(arg) > 0L && !rlang::is_named(arg)
> ```
> 
> the `rlang::` should be removed.

@bundfussr please feel free to push small changes like this

**Comment by bundfussr** [comment]
> > In line 965 of `assertions.R`
> > ```
> > any_unnamed <- length(arg) > 0L && !rlang::is_named(arg)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the `rlang::` should be removed.
> 
> @bundfussr please feel free to push small changes like this

Done

**Review by bundfussr — APPROVED** [review]

**Pull request merged**